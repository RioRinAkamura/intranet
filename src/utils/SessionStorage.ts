export class SessionStorage {
  // store key/value pair
  public async set(key: string, value: string): Promise<void> {
    localStorage.setItem(key, value);
  }

  // get value of
  public async get(key: string): Promise<string> {
    const value = localStorage.getItem(key);
    return value || '';
  }

  // delete key
  public async remove(key: string): Promise<void> {
    localStorage.removeItem(key);
  }
}
