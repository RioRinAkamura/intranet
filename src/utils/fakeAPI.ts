// First we need to import axios.js
import axios from 'axios';
// Next we make an 'instance' of it
const instance = axios.create({
  // .. where we make our configurations
  baseURL: 'https://api.boilerplate.dev.hdwebsoft.co/v1',
});

// Also add/ configure interceptors && all the other cool stuff

instance.interceptors.request.use(
  config => {
    if (!config.headers.Authorization) {
      const auth = localStorage.getItem('AUTH_SESSION_KEY');
      if (auth) {
        const token = JSON.parse(auth).access;
        if (token) {
          config.headers.Authorization = `Bearer ${token}`;
        }
      }
    }

    return config;
  },
  error => {
    console.log(error);
    return Promise.reject(error);
  },
);

instance.interceptors.response.use(
  response => {
    // Edit response config
    return response.data;
  },
  error => {
    console.log(error);
    return Promise.reject(error);
  },
);

export default instance;
