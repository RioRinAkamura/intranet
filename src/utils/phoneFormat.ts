export function phoneFormat(phone: string): string {
  if (phone.length === 10) {
    return phone.replace(/(\d{3})(\d{3})(\d{4})/, `$1 $2 $3`);
  } else if (phone.length === 11) {
    return phone.replace(/(\d{4})(\d{3})(\d{4})/, `$1 $2 $3`);
  } else {
    // This is temporary, if there are other phone format, add more
    return phone;
  }
}
