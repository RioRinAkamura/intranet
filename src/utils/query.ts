import querystring from 'query-string';
export function getQueryParam<T>(): T {
  if (typeof window === 'undefined') return {} as T;
  return querystring.parse(window.location.search) as any;
}