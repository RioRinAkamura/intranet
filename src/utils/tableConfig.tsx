import * as React from 'react';
import { has } from 'lodash';
import {
  Button,
  Checkbox,
  CheckboxOptionType,
  Col,
  Input,
  Row,
  Space,
  Select,
  Modal,
} from 'antd';
import { SearchOutlined } from '@ant-design/icons';
import { useTranslation } from 'react-i18next';
import Highlighter from 'react-highlight-words';
import { TagsInput } from 'app/components/Tags';
import { TableStateProps } from 'app/pages/UserPage/UserListPage/useHandleDataTable';
import { MessageTranslate, TagType } from './types';
import styled from 'styled-components/macro';
import { FilterColumns } from 'app/pages/UserPage/UserListPage/slice/types';
import { useProjectDetail } from 'app/pages/ProjectPage/ProjectDetailPage/useProjectDetail';

interface useTableProps {
  getColumnSorterProps: (dataIndex: string, columnPriority: number) => {};
  getColumnSearchInputProps: (dataIndex: string[], filterIndex?: number) => {};
  getColumnSearchTagProps: (dataIndex: string, tags?: TagType[]) => {};
  getColumnSearchCheckboxProps: (
    dataIndex: string[],
    options: CheckboxOptionType[],
    filterIndex?: number,
  ) => {};
  getColumnSearchCheckboxFromToProps: (
    dataIndex: string[],
    options: CheckboxOptionType[],
    range: string,
    filterIndex?: number,
  ) => {};
  ConfirmModal: () => JSX.Element;
}

const { Option } = Select;

export const useTableConfig = (
  state: TableStateProps,
  messageTrans: MessageTranslate,
  setFilterText: (value: FilterColumns) => void,
): useTableProps => {
  const { t } = useTranslation();
  const { update } = useProjectDetail();
  const [selectedKeys, setSelectedKeys] = React.useState([]);
  const [visible, setVisible] = React.useState(false);
  const [formValue, setFormValue] = React.useState({});

  React.useLayoutEffect(() => {
    if (state.filterColumns) {
      setSelectedKeys(prev => ({ ...prev, ...state.filterColumns }));
    }
  }, [state.filterColumns]);

  const handleOkConfirmModal = async () => {
    try {
      await update(formValue);
      setVisible(false);
    } catch (e) {
      console.log(e);
    }
  };

  const handleCancelConfirmModal = () => {
    setVisible(false);
  };

  const ConfirmModal = () => (
    <Modal
      title="Confirm Modal"
      visible={visible}
      onOk={handleOkConfirmModal}
      onCancel={handleCancelConfirmModal}
      okText="Ok"
      cancelText="Cancel"
    >
      <p>Are you sure this project has been completed?</p>
    </Modal>
  );

  const getColumnSorterProps = (dataIndex: string, columnPriority: number) => {
    const ordering = {
      sorter: {
        multiple: columnPriority,
      },
    };
    if (state.params.ordering) {
      ordering['sortOrder'] = state.params.ordering.includes(dataIndex)
        ? state.params.ordering.includes('-' + dataIndex)
          ? ('descend' as 'descend')
          : ('ascend' as 'ascend')
        : null;
    } else if (state.params.ordering === '') {
      ordering['sortOrder'] = null;
    }
    return ordering;
  };

  const getColumnSearchInputProps = (
    dataIndex: string[],
    filterIndex?: number,
  ) => ({
    ellipsis: true,
    filterDropdown: ({ confirm }) => {
      return (
        <Wrapper>
          <Input
            placeholder={`${t(
              messageTrans.filterInputPlaceholder(),
            )} ${dataIndex}`}
            value={selectedKeys[dataIndex[filterIndex || 0]]}
            onChange={e => {
              e.persist();
              setSelectedKeys(prevState => ({
                ...prevState,
                [dataIndex[filterIndex || 0]]: e.target.value
                  ? e.target.value
                  : null,
              }));
            }}
            onPressEnter={() =>
              handleSearch(dataIndex[filterIndex || 0], confirm)
            }
            style={{ width: 188, marginBottom: 8, display: 'block' }}
          />
          <Space>
            <Button
              type="primary"
              onClick={() => handleSearch(dataIndex[filterIndex || 0], confirm)}
              icon={<SearchOutlined />}
              size="small"
              style={{ width: 90 }}
              loading={state.loading}
            >
              {t(messageTrans.filterSearchButton())}
            </Button>
            <Button
              onClick={() => handleReset(dataIndex[filterIndex || 0], confirm)}
              size="small"
              loading={state.loading}
              style={{ width: 90 }}
            >
              {t(messageTrans.filterResetButton())}
            </Button>
          </Space>
        </Wrapper>
      );
    },
    filterIcon: filtered => (
      <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />
    ),
    onFilter: (value, record) =>
      record[dataIndex[filterIndex || 0]]
        ? record[dataIndex[filterIndex || 0]]
            .toString()
            .toLowerCase()
            .includes(value.toLowerCase())
        : '',
    render: (text, record) => {
      let dataText = '';
      dataIndex.map(data => {
        if (record[data]) {
          dataText += record[data] + ' ';
        }
        return data;
      });
      return has(state.filterColumns, dataIndex[filterIndex || 0]) ||
        (state.params.search && state.params.search.length > 0) ? (
        <Highlighter
          highlightStyle={{ backgroundColor: '#ffc069', padding: 0 }}
          searchWords={[
            state.filterColumns![dataIndex[filterIndex || 0]],
            state.params.search &&
              state.params.search.length > 0 &&
              state.params.search,
          ]}
          autoEscape
          textToHighlight={text ? dataText.trim().toString() : ''}
        />
      ) : (
        dataText.trim()
      );
    },
  });

  const handleSearch = (dataIndex: string, confirm: () => void) => {
    setFilterText({ [dataIndex]: selectedKeys[dataIndex] });
    confirm();
  };

  const handleReset = (dataIndex: string, confirm: () => void) => {
    setSelectedKeys(prevState => ({
      ...prevState,
      [dataIndex]: undefined,
    }));
    setFilterText({ [dataIndex]: undefined });
    confirm();
  };

  const handleFromToSearch = (
    dataIndex: string,
    range: string,
    confirm: () => void,
  ) => {
    if (selectedKeys[dataIndex] === '1') {
      setFilterText({ to: range, from: undefined });
    } else if (selectedKeys[dataIndex] === '2') {
      setFilterText({ from: range, to: range });
    } else {
      setFilterText({ from: range, to: undefined });
    }
    confirm();
  };

  const handleFromToReset = (dataIndex: string, confirm: () => void) => {
    setSelectedKeys(prevState => ({
      ...prevState,
      [dataIndex]: undefined,
      from: undefined,
      to: undefined,
    }));
    setFilterText({ from: undefined, to: undefined });
    confirm();
  };

  const getColumnSearchTagProps = (dataIndex: string) => ({
    filterDropdown: ({ confirm }) => {
      return (
        <Wrapper>
          <TagsInput
            value={selectedKeys[dataIndex]}
            callback={e => {
              setSelectedKeys(prevState => ({
                ...prevState,
                [dataIndex]: e ? e : null,
              }));
            }}
          />
          <Space>
            <Button
              type="primary"
              onClick={() => handleSearch(dataIndex, confirm)}
              icon={<SearchOutlined />}
              size="small"
              style={{ width: 90 }}
              loading={state.loading}
            >
              {t(messageTrans.filterSearchButton())}
            </Button>
            <Button
              onClick={() => handleReset(dataIndex, confirm)}
              size="small"
              loading={state.loading}
              style={{ width: 90 }}
            >
              {t(messageTrans.filterResetButton())}
            </Button>
          </Space>
        </Wrapper>
      );
    },
    filterIcon: filtered => (
      <SearchOutlined style={{ color: filtered ? '#1890ff' : undefined }} />
    ),
    onFilter: (value, record) =>
      record[dataIndex]
        ? record[dataIndex]
            .toString()
            .toLowerCase()
            .includes(value.toLowerCase())
        : '',
  });

  const getColumnSearchCheckboxProps = (
    dataIndex: string[],
    options: CheckboxOptionType[],
    filterIndex?: number,
  ) => ({
    filterDropdown: ({ confirm }) => {
      return (
        <Wrapper>
          <WrapperCheckbox>
            <Checkbox.Group
              value={selectedKeys[dataIndex[filterIndex || 0]]}
              options={options}
              onChange={e => {
                setSelectedKeys(prevState => ({
                  ...prevState,
                  [dataIndex[filterIndex || 0]]: e ? e : null,
                }));
              }}
            />
          </WrapperCheckbox>
          <Row gutter={[8, 0]}>
            <Col>
              <Button
                type="primary"
                onClick={() =>
                  handleSearch(dataIndex[filterIndex || 0], confirm)
                }
                icon={<SearchOutlined />}
                size="small"
                style={{ width: 90 }}
                loading={state.loading}
              >
                {t(messageTrans.filterSearchButton())}
              </Button>
            </Col>
            <Col>
              <Button
                onClick={() =>
                  handleReset(dataIndex[filterIndex || 0], confirm)
                }
                size="small"
                loading={state.loading}
                style={{ width: 90 }}
              >
                {t(messageTrans.filterResetButton())}
              </Button>
            </Col>
          </Row>
        </Wrapper>
      );
    },
    onFilter: (value, record) =>
      record[dataIndex[filterIndex || 0]]
        ? record[dataIndex[filterIndex || 0]]
            .toString()
            .toLowerCase()
            .includes(value.toLowerCase())
        : '',
    render: (text, record) => {
      let dataText = '';
      dataIndex.map(data => {
        if (record[data]) {
          dataText += record[data] + ' ';
        }
        return data;
      });
      const handleValueChange = async value => {
        const defaultForm = {
          ...record,
        };
        defaultForm[dataIndex[0]] = value;

        // hard code
        if (dataIndex[0] === 'status' && value === '4') {
          setVisible(true);
          setFormValue({ ...defaultForm });
          return;
        }

        try {
          await update(defaultForm);
        } catch (e) {
          console.log(e);
        }
      };
      let defaultValue: string = '';
      const findOption = options.find(
        option => option.value === Number(dataText),
      );
      defaultValue = findOption
        ? findOption.label
          ? findOption.label.toString()
          : ''
        : '';
      return has(state.filterColumns, dataIndex[filterIndex || 0]) ||
        (state.params.search && state.params.search.length > 0) ? (
        <Select
          onChange={handleValueChange}
          defaultValue={defaultValue}
          style={{ width: '100%' }}
        >
          {options.map(option => {
            return (
              <Option value={`${option.value}`}>
                <Highlighter
                  highlightStyle={{ backgroundColor: '#ffc069', padding: 0 }}
                  searchWords={[
                    state.filterColumns![dataIndex[filterIndex || 0]]?.includes(
                      text,
                    ) &&
                      options.find(option => option.value === Number(text))
                        ?.label,
                    state.params.search &&
                      state.params.search.length > 0 &&
                      state.params.search,
                  ]}
                  autoEscape
                  textToHighlight={option.label}
                />
              </Option>
            );
          })}
        </Select>
      ) : (
        <Select
          onChange={handleValueChange}
          defaultValue={defaultValue}
          style={{ width: '100%' }}
        >
          {options.map(option => {
            return <Option value={`${option.value}`}>{option.label}</Option>;
          })}
        </Select>
        // options.find(option => option.value === Number(dataText))?.label
      );
    },
  });

  const getColumnSearchCheckboxFromToProps = (
    dataIndex: string[],
    options: CheckboxOptionType[],
    range: string,
    filterIndex?: number,
  ) => ({
    filterDropdown: ({ confirm }) => {
      return (
        <Wrapper>
          <WrapperCheckbox>
            <Checkbox.Group
              value={selectedKeys[dataIndex[filterIndex || 0]]}
              options={options}
              onChange={e => {
                setSelectedKeys(prevState => ({
                  ...prevState,
                  [dataIndex[filterIndex || 0]]: e.toString()
                    ? e.toString()
                    : null,
                }));
              }}
            />
          </WrapperCheckbox>
          <Row gutter={[8, 0]}>
            <Col>
              <Button
                type="primary"
                onClick={() =>
                  handleFromToSearch(
                    dataIndex[filterIndex || 0],
                    range,
                    confirm,
                  )
                }
                icon={<SearchOutlined />}
                size="small"
                style={{ width: 90 }}
                loading={state.loading}
              >
                {t(messageTrans.filterSearchButton())}
              </Button>
            </Col>
            <Col>
              <Button
                onClick={() =>
                  handleFromToReset(dataIndex[filterIndex || 0], confirm)
                }
                size="small"
                loading={state.loading}
                style={{ width: 90 }}
              >
                {t(messageTrans.filterResetButton())}
              </Button>
            </Col>
          </Row>
        </Wrapper>
      );
    },
    onFilter: (value, record) =>
      record[dataIndex[filterIndex || 0]]
        ? record[dataIndex[filterIndex || 0]]
            .toString()
            .toLowerCase()
            .includes(value.toLowerCase())
        : '',
  });

  return {
    getColumnSorterProps,
    getColumnSearchInputProps,
    getColumnSearchTagProps,
    getColumnSearchCheckboxProps,
    getColumnSearchCheckboxFromToProps,
    ConfirmModal,
  };
};

const Wrapper = styled.div`
  padding: 8px;
`;

const WrapperCheckbox = styled.div`
  .ant-checkbox-group {
    display: grid;
  }

  border-bottom: 1px solid #d5d4d5;
  padding-bottom: 7px;
  margin-bottom: 7px;
`;
