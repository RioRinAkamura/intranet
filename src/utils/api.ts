import { Api, createApiClient } from '@hdwebsoft/boilerplate-api-sdk';
import { SessionStorage } from './SessionStorage';

export const sessionStorage = new SessionStorage();
const config = {
  baseUrl: 'https://api.boilerplate.dev.hdwebsoft.co/v1',
  authSessionKey: 'AUTH_SESSION_KEY',
  session: sessionStorage,
  socketUrl: 'http://template-ws.dev.goldfishcode.com',
};

const client = createApiClient(config);
export const api = new Api(client);
