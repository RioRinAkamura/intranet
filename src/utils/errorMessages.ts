import { translations } from 'locales/translations';
import { _t } from 'utils/messages';

export interface ErrorType {
  code: number;
  message: string;
  status_code: number;
  summary: string;
}

export const errorMess = {
  400: () => _t(translations.Errors.error400, 'Bad Request'),
  401: () => _t(translations.Errors.error401, 'Unauthorized'),
  403: () => _t(translations.Errors.error403, 'Forbidden'),
  404: () => _t(translations.Errors.error404, 'Not Found'),
  500: () => _t(translations.Errors.error500, 'Internal Server Error'),
};

export const mapErrorCode = (error: ErrorType) => {
  if (error) {
    const { code, message } = error;
    return errorMess[code] ? errorMess[code] : message;
  }
};
