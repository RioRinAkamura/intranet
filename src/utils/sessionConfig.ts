import { Api, createApiClient, SocketIO } from '@goldfishcode/template-api-sdk';
import { SessionStorage } from './SessionStorage';

export const sessionStorage = new SessionStorage();
const config = {
  baseUrl: 'https://template-api.dev.goldfishcode.com/v1',
  authSessionKey: 'AUTH_SESSION_KEY',
  session: sessionStorage,
  socketUrl: 'http://template-ws.dev.goldfishcode.com',
};

const client = createApiClient(config);
export const api = new Api(client);
