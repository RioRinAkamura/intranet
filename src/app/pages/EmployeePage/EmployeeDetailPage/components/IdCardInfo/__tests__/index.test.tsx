import * as React from 'react';
import { render } from '@testing-library/react';

import { IdCardInfo } from '..';
import { matchMedia } from 'utils/matchMedia';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

matchMedia();

describe('<IdCardInfo  />', () => {
  it('should match snapshot', () => {
    const loadingIndicator = render(
      <IdCardInfo isView={true} isEdit={false} form={undefined} />,
    );
    expect(loadingIndicator.container.firstChild).toMatchSnapshot();
  });
});
