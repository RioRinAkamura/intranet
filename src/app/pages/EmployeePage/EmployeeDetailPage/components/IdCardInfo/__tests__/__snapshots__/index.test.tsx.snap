// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IdCardInfo  /> should match snapshot 1`] = `
.c1 {
  color: #1890ff;
  margin-bottom: 24px;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0px;
}

.c2 div {
  width: 100%;
}

.c2 label {
  font-weight: 500;
}

.c0 .label {
  font-weight: 500;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <b>
       
      UserDetailPage.modal.form.profile.idCard.title
      ID card information
    </b>
  </div>
  <div
    class="ant-row"
    style="margin-left: -16px; margin-right: -16px;"
  >
    <div
      class="ant-col ant-col-xs-24 ant-col-md-24"
      style="padding-left: 16px; padding-right: 16px;"
    >
      <div
        class="ant-row ant-row-middle"
        style="margin-top: -6px; margin-bottom: -6px;"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-md-8"
          style="padding-top: 6px; padding-bottom: 6px;"
        >
          <span
            class="label"
          >
            UserDetailPage.modal.form.profile.idNumber.label
            ID number
          </span>
        </div>
        <div
          class="ant-col ant-col-xs-24 ant-col-md-16"
          style="padding-top: 6px; padding-bottom: 6px;"
        >
          <div
            class="ant-row ant-form-item c2"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input ant-input-lg ant-input-borderless"
                    id="id_number"
                    placeholder=""
                    readonly=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-24 ant-col-md-24"
      style="padding-left: 16px; padding-right: 16px;"
    >
      <div
        class="ant-row ant-row-middle"
        style="margin-top: -6px; margin-bottom: -6px;"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-md-8"
          style="padding-top: 6px; padding-bottom: 6px;"
        >
          <span
            class="label"
          >
            UserDetailPage.modal.form.profile.issuedDate.label
            Issued date
          </span>
        </div>
        <div
          class="ant-col ant-col-xs-24 ant-col-md-16"
          style="padding-top: 6px; padding-bottom: 6px;"
        >
          <div
            class="ant-row ant-form-item c2"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <div
                    class="ant-picker ant-picker-large ant-picker-borderless"
                  >
                    <div
                      class="ant-picker-input"
                    >
                      <input
                        autocomplete="off"
                        id="issued_date"
                        placeholder=""
                        readonly=""
                        size="12"
                        title=""
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-col ant-col-xs-24 ant-col-md-24"
      style="padding-left: 16px; padding-right: 16px;"
    >
      <div
        class="ant-row ant-row-middle"
        style="margin-top: -6px; margin-bottom: -6px;"
      >
        <div
          class="ant-col ant-col-xs-24 ant-col-md-8"
          style="padding-top: 6px; padding-bottom: 6px;"
        >
          <span
            class="label"
          >
            UserDetailPage.modal.form.profile.issuedPlace.label
            Issued place
          </span>
        </div>
        <div
          class="ant-col ant-col-xs-24 ant-col-md-16"
          style="padding-top: 6px; padding-bottom: 6px;"
        >
          <div
            class="ant-row ant-form-item c2"
          >
            <div
              class="ant-col ant-form-item-control"
            >
              <div
                class="ant-form-item-control-input"
              >
                <div
                  class="ant-form-item-control-input-content"
                >
                  <input
                    class="ant-input ant-input-lg ant-input-borderless"
                    id="issued_place"
                    placeholder=""
                    readonly=""
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
